python/ml4ir/tests/test_serving.py

```diff
@@ -87,32 +87,40 @@ def get_dataset(parse_tfrecord):
         parsed_dataset_batch = parsed_dataset.test.take(1)
 
         # Use the loaded serving signatures for inference
-        model_predictions = model.predict(parsed_dataset_batch)
+        model_predictions = model.predict(parsed_dataset_batch)["new_score"].values
         default_signature_predictions = default_signature(**parsed_sequence_examples)[
             "ranking_scores"
         ]
         tfrecord_signature_predictions = tfrecord_signature(
             sequence_example_protos=sequence_example_protos
         )["ranking_scores"]
 
+        @tf.function
```

Comment:
tf.functions inside of test classes?  Can you write up what these are doing?

Reply:
They don't _have to_ be `tf.function` actually. Can remove if you think it is inappropriate. I am trying to replicate the same flatten and filtering out padded records that I do in the `.predict()` method of `RankingModel` to the signatures as well in order to unit test that the predictions from all serving signatures + the actual keras model is the same.

Reply:
you don't need to remove it, but it's confusing that there are functions meant to be wired into the graph, and yet we're in the test classes...

You're writing code which replicates what is done with padding in the main model, to test it.  I think it's just hard for me to read this to be sure I understand the logic we "know" is correct.